modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1577,7 +1577,7 @@ def describe(self, **kwargs):
 
         def describe_builder(df, internal_indices=[]):
             """Apply `describe` function to the subset of columns in a single partition."""
-            return df.iloc[:, internal_indices].describe(**kwargs)
+            return df.iloc[:, internal_indices].describe(**kwargs).reindex(new_index)
```

Comment:
There may be some performance implications here but I presume that it should be ok given that we're working at partition granularity?

Reply:
Thanks for running the quick benchmark as a sanity. Yeah, we need the extra rows either way, and I don't know of an easier way of doing that then with `reindex` either. I think the time of the reindex will also get larger pretty quickly with the number of additional rows we add. Hopefully it won't be terribly large in the case of `describe`.

Reply:
We could potentially mitigate the performance hit by checking if the index of df.describe equals the desired index and only reinforcing if it doesn't?

Reply:
@RehanSD is there a performance penalty for calling reindex if the index is already the same?

Reply:
@RehanSD I thought about that, but in practice it looks like the time to `describe` outweights the time to `reindex` with an equal index by a factor of over 1000:

```python
import pandas as pd
import numpy as np
np.random.seed(0)
df = pd.DataFrame(np.random.randint(0,100,(1,2 ** 12)))
%time df2 = df.describe()
 %time df2 = df2.reindex(df2.index)
```

I get 4.56 seconds for the describe and 282 microseconds for the reindex, so I don't think the extra optimization is worth the extra complexity.

Reply:
The docs say this: "A new object is produced unless the new index is equivalent to the current one and copy=False". In this case, copy is not False, so maybe the right thing to do here is to make copy=False?

Reply:
Yeah I just tried experimenting a bit with this case and came to the same conclusion.